packages/core/src/database/index.ts

```diff
@@ -0,0 +1,1211 @@
+            }
+
+            /**
+             * If cache hit, start
+             *
+             * A cache hit occurs if the previous app has the same build id
+             * as the new app. In this case, we can remove indexes, revert
+             * unfinalized data and continue where it left off.
+             */
+            if (
+              args.common.options.command !== "dev" &&
+              previousApp.build_id === buildId &&
+              previousApp.checkpoint !== encodeCheckpoint(zeroCheckpoint)
+            ) {
+              await tx
+                .updateTable("_ponder_meta")
+                .set({
+                  value: encodeApp({
+                    ...previousApp,
+                    is_locked: true,
+                    is_dev: false,
+                    heartbeat_at: Date.now(),
+                  }),
+                })
+                .where("key", "=", "app")
+                .execute();
+
+              args.common.logger.info({
+                service: "database",
+                msg: `Detected cache hit for build '${buildId}' in schemaq '${namespace}' last active ${formatEta(Date.now() - previousApp.heartbeat_at)} ago`,
+              });
+              args.common.logger.debug({
+                service: "database",
+                msg: `Acquired lock on schema '${namespace}'`,
+              });
+
+              // Remove indexes
+              for (const [tableName, table] of Object.entries(
+                getTables(args.schema),
+              )) {
+                if (table.constraints === undefined) continue;
+
+                for (const name of Object.keys(table.constraints)) {
+                  await tx.schema
+                    .dropIndex(`${tableName}_${name}`)
+                    .ifExists()
+                    .execute();
+
+                  args.common.logger.info({
+                    service: "database",
+                    msg: `Dropped index '${tableName}_${name}' in schema '${namespace}'`,
+                  });
+                }
+              }
+
+              // Revert unfinalized data
+
+              const { blockTimestamp, chainId, blockNumber } = decodeCheckpoint(
+                previousApp.checkpoint,
+              );
+              args.common.logger.info({
+                service: "database",
+                msg: `Reverting operations prior to finalized checkpoint (timestamp=${blockTimestamp} chainId=${chainId} block=${blockNumber})`,
+              });
+
+              for (const tableName of Object.keys(getTables(args.schema))) {
+                await revert({
+                  tableName,
+                  checkpoint: previousApp.checkpoint,
+                  tx,
+                });
+              }
+
+              return {
+                status: "success",
+                checkpoint: previousApp.checkpoint,
+              } as const;
+            }
+
+            /**
+             * If table name collision, exit
+             * Else, start
```

Comment:
Is this comment correct? Looks like you may have forgotten to write the table name collision logic.